lib/ansible/plugins/inventory/linode.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+    name: linode
+    plugin_type: inventory
+    authors:
+      - Luke Murphy (@lwm)
+    short_description: Ansible dynamic inventory plugin for Linode.
+    version_added: "2.8"
+    requirements:
+        - python >= 2.7
+        - linode_api4 >= 2.0.0
+    description:
+        - Reads inventories from the Linode API v4.
+        - Uses a <name>.linode.yaml (or <name>.linode.yml) YAML configuration file.
+        - Linode labels are used by default as the hostnames.
+        - The inventory groups are built from groups and not tags.
+    options:
+        plugin:
+            description: marks this as an instance of the 'linode' plugin
+            required: true
+            choices: ['linode']
+        access_token:
+            description: The Linode account personal access token.
+            required: true
+            env:
+                - name: LINODE_ACCESS_TOKEN
+        regions:
+          description: Populate inventory with instances in this region.
+          default: []
+          type: list
+          required: false
+        types:
+          description: Populate inventory with instances with this type.
+          default: []
+          type: list
+          required: false
+'''
+
+EXAMPLES = r'''
+# Minimal example. `LINODE_ACCESS_TOKEN` is exposed in environment.
+plugin: linode
+
+# Example with regions, types, groups and access token
+plugin: linode
+access_token: foobar
+regions:
+  - eu-west
+types:
+  - g5-standard-2
+'''
+
+import os
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.module_utils.six import string_types
+from ansible.plugins.inventory import BaseInventoryPlugin
+
+
+try:
+    from linode_api4 import LinodeClient
+    from linode_api4.errors import ApiError as LinodeApiError
+except ImportError:
+    raise AnsibleError('the Linode dynamic inventory plugin requires linode_api4.')
```

Comment:
I encountered this error testing with `ansible-inventory`.  It was only visible with `-v`

Reply:
:+1: